pkg/service/cluster/service.go

```diff
@@ -462,42 +464,44 @@ func (s *Service) PutCluster(ctx context.Context, c *Cluster) (err error) {
 }
 
 func (s *Service) validateHostsConnectivity(ctx context.Context, c *Cluster) error {
-	// If host changes ignore old known hosts.
-	if c.Host != "" {
-		c.KnownHosts = []string{c.Host}
-	} else if err := s.loadKnownHosts(c); err != nil {
+	if err := s.loadKnownHosts(c); err != nil && !errors.Is(err, gocql.ErrNotFound) {
 		return errors.Wrap(err, "load known hosts")
 	}
+	if c.Host != "" {
+		c.KnownHosts = append([]string{c.Host}, c.KnownHosts...)
+	}
 
-	client, err := s.createClientNoValidation(c)
+	knwonHosts, err := s.filterKnownHosts(ctx, c)
 	if err != nil {
-		return errors.Wrap(err, "create client")
+		if errors.Is(err, ErrNoValidKnownHost) {
+			s.logger.Error(ctx, "There is no single valid known host for the cluster."+
+				"Please update it with 'sctool cluster update -h <host>'.", "cluster", c.ID, "coordinatorHost", c.Host,
+				"knownHosts", c.KnownHosts)
+		}
+		return err
 	}
-	defer logutil.LogOnError(ctx, s.logger, client.Close, "Couldn't close scylla client")
+	c.KnownHosts = knwonHosts
 
-	status, err := client.Status(ctx)
+	config := s.clientConfig(c)
+	client, err := scyllaclient.NewClient(config, s.logger.Named("client"))
 	if err != nil {
-		return errors.Wrap(err, "status")
+		return err
 	}
+	defer logutil.LogOnError(ctx, s.logger, client.Close, "Couldn't close scylla client")
 
-	// Get live hosts
+	status, err := client.Status(ctx)
 	live := status.Live().Hosts()
 	if len(live) == 0 {
 		return service.ErrValidate(errors.New("no live nodes"))
 	}
 
-	// For every live host check there are no errors
 	var errs error
 	for i, err := range client.CheckHostsConnectivity(ctx, live) {
 		errs = multierr.Append(errs, errors.Wrap(err, live[i]))
 	}
 	if errs != nil {
 		return service.ErrValidate(errors.Wrap(errs, "connectivity check"))
 	}
-
-	// Update known hosts.
-	c.KnownHosts = live
```

Comment:
So now we set known hosts to all known hosts, not only to the live ones.
This could result in some additional error logs, but client should be able to handle down nodes in some way.

Reply:
known hosts are the ones discovered with the API call, not the ones that are live only